### PR TITLE
Aizad/FEQ-1502/Adding New Variants inside of Button

### PR DESCRIPTION
**Pull request by aizad-deriv**
- Refactor Button components
- Updated styles for Button components
- Button components supports variants in all different colors
- Updated Storybook with the changes

image

### CODE REVIEW

File: lib/components/Button/Button.scss [hunk 1]
```diff
@@ -111,3 +82,152 @@ $size-map: (
...
+    &.deriv-button__color--black {
+        background-color: $black;
+        span {
+            color: $white;
+        }
+        &:hover:not(:disabled) {
+            background-color: $black-hover;
+        }
+    }
+}
```
Comment (ameerul-deriv) [inline lib/components/Button/Button.scss:134]: can we wrap all of the individual variants inside each of the parent variants? something like this?

```suggestion
.deriv-button__variant--contained {
    &.deriv-button__color--primary {
        background-color: $primary;
        span {
            color: $white;
        }
        &:hover:not(:disabled) {
            background-color: $primary-hover;
        }
    }
    
    &.deriv-button__color--primary-light {
        background-color: $primary-light;
        span {
            color: $primary;
        }
        &:hover:not(:disabled) {
            background-color: $primary-light-hover;
        }
    }
    
     &.deriv-button__color--white {
        background-color: $white;
        span {
            color: $black;
        }
        &:hover:not(:disabled) {
            background-color: $white-hover;
        }
    }
    
    &.deriv-button__color--black {
        background-color: $black;
        span {
            color: $white;
        }
        &:hover:not(:disabled) {
            background-color: $black-hover;
        }
    }
}
```

Comment (ameerul-deriv) [reply lib/components/Button/Button.scss:134]: same with the others below 👀

Comment (rupato-deriv) [reply lib/components/Button/Button.scss:134]: @aizad-deriv can we use this ?
`@mixin deriv-button-contained($background-color, $text-color, $hover-background-color) {
  background-color: $background-color;
  span {
    color: $text-color;
  }
  &:hover:not(:disabled) {
    background-color: $hover-background-color;
  }
}

.deriv-button__variant--contained {
  &.deriv-button__color--primary {
    @include deriv-button-contained($primary, $white, $primary-hover);
  }

  &.deriv-button__color--primary-light {
    @include deriv-button-contained($primary-light, $primary, $primary-light-hover);
  }

  &.deriv-button__color--white {
    @include deriv-button-contained($white, $black, $white-hover);
  }
}`

File: src/stories/Button.stories.ts [hunk 2]
```diff
@@ -117,3 +117,75 @@ export const ContainedWhite: Story = {
...
+};
+
+export const GhostWhite: Story = {
+    name: "Ghost (White)",
+    args: {
+        variant: "ghost",
+        children: "Button",
+        color: "white",
+    },
+};
```
Comment (rupato-deriv) [inline src/stories/Button.stories.ts:191]: same here! 

`const createButtonStory = (name, variant, color) => ({
  name: `${variant.charAt(0).toUpperCase() + variant.slice(1)} (${color.charAt(0).toUpperCase() + color.slice(1)})`,
  args: {
    variant,
    children: "Button",
    color,
  },
});

export const OutlinedPrimary = createButtonStory("Outlined", "outlined", "primary");
export const OutlinedPrimaryLight = createButtonStory("Outlined", "outlined", "primary-light");
export const OutlinedBlack = createButtonStory("Outlined", "outlined", "black");
export const OutlinedWhite = createButtonStory("Outlined", "outlined", "white");

export const GhostPrimary = createButtonStory("Ghost", "ghost", "primary");
export const GhostPrimaryLight = createButtonStory("Ghost", "ghost", "primary-light");
export const GhostBlack = createButtonStory("Ghost", "ghost", "black");
export const GhostWhite = createButtonStory("Ghost", "ghost", "white");`

Comment (aizad-deriv) [reply src/stories/Button.stories.ts:180]: I rather take the argument from `meta` and replace it with the values in each of the `Story`.

**Review by ameerul-deriv — APPROVED** [review]

**Review by thisyahlen-deriv — APPROVED** [review]

File: lib/components/Button/Button.scss [hunk 3]
```diff
@@ -1,44 +1,21 @@
...
+$white-outline-hover: #ffffff14;
+$black-outline-hover: rgba(0, 0, 0, 0.08);
 
 .deriv-button {
     display: inline-flex;
     justify-content: center;
     align-items: center;
     position: relative;
-    gap: 8px;
+    gap: 0.8rem;
```
Comment (niloofar-deriv) [inline lib/components/Button/Button.scss:18]: Shouldn't be px?

File: lib/components/Button/Button.scss [hunk 4]
```diff
@@ -47,38 +24,147 @@ $size-map: (
         cursor: not-allowed;
     }
 
-    @each $size, $values in $size-map {
-        &__size--#{$size} {
-            padding: map-get($values, padding);
-            height: map-get($values, height);
+    &__size {
+        &--sm {
+            height: 2.4rem;
```
Comment (niloofar-deriv) [inline lib/components/Button/Button.scss:29]: same

File: lib/components/Button/Button.scss [hunk 5]
```diff
@@ -47,38 +24,147 @@ $size-map: (
...
+            border-width: 1px;
+            padding: 0.3rem 0.8rem;
```
Comment (niloofar-deriv) [inline lib/components/Button/Button.scss:31]: same

File: lib/components/Button/Button.scss [hunk 6]
```diff
@@ -47,38 +24,147 @@ $size-map: (
...
-    }
-
-    @each $color, $values in $color-map {
-        &__color--#{$color} {
-            background-color: map-get($values, backgroundcolor);
-            &:hover:not(:disabled) {
-                background-color: map-get($values, hover);
-            }
+        &--md {
+            height: 3.2rem;
```
Comment (niloofar-deriv) [inline lib/components/Button/Button.scss:34]: same

**Review by niloofar-deriv — APPROVED** [review]
Base on our discussion we will investigate more on using rem or px later.
The rest is LGTM 👍

File: lib/components/Button/Button.scss (hunk 3 above)
Comment (shayan-deriv) [reply lib/components/Button/Button.scss:18]: @niloofar-deriv we are changing them to `rem` from now on. am I right @aizad-deriv ?

Comment (niloofar-deriv) [reply lib/components/Button/Button.scss:18]: Right @shayan-deriv

**Review by shayan-deriv — APPROVED** [review]

**Pull request merged**